lib/src/interfaces/NavigationFunctionComponent.ts

```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { NavigationComponentProps } from './NavigationComponentProps';
+import { Options } from './Options';
+
+export interface NavigationFunctionComponent<Props = {}>
+  extends React.FunctionComponent<Props & NavigationComponentProps> {
+    options?: Options;
```

Comment:
Correct me if I'm wrong but I believe `options` can also be a function with props. In my project, I have it typed as:

```tsx
options?: ((props: Props) => Options) | Options
```

So you could do:

```tsx
const Component: NavigationFunctionComponent<{ name: string }> = () => {}

Component.options = ({ name }) => ({
  topBar: {
    title: {
      text: name
    }
  }
})
```

Reply:
That's true, as stated in [the docs](https://wix.github.io/react-native-navigation/docs/passing-data-to-components#handling-passprops-in-static-options). I'm not used to defining options that way.

I've checked that `NavigationComponentProps` are not passed to options functions. Interface fixed and rebased onto master.